src/routes/login_page.jsx

```diff
@@ -17,54 +18,77 @@ export default class LoginPage extends React.Component {
       isNetworkError: false,
     };
 
-    this.handleInputChange = this.handleInputChange.bind(this);
-    this.handleLoginSubmit = this.handleLoginSubmit.bind(this);
+    this.updateField = this.updateField.bind(this);
     this.handleAlertClose = this.handleAlertClose.bind(this);
+    this.handleLogin = this.handleLogin.bind(this);
   }
 
-  handleInputChange(e, field) {
+  updateField(e, field) {
     this.setState({
-      [field]: e.target.value,
+      [field]: e.currentTarget.value,
     });
   }
 
-  handleLoginSubmit(event) {
-    const { username, password } = this.state;
-    event.preventDefault();
-    if (!(username && password)) {
+  async handleLogin(e) {
+    e.preventDefault();
+    const { email, password } = this.state;
+
+    if (!(email || password)) {
       this.setState({
         isAuthenticationError: true,
         isInvalidInput: true,
         isInvalidCredentials: false,
         isNetworkError: false,
       });
-
       return;
     }
 
     this.setState({
-      isAuthenticating: false,
-      isAuthenticated: false,
-      isAuthenticationError: false,
-      isInvalidCredentials: false,
-      isInvalidInput: false,
-      isNetworkError: false,
+      isAuthenticating: true,
     });
 
-    if (username === "ziv" && password === "papa") {
-      this.setState({
-        isAuthenticating: false,
-        isAuthenticated: true,
-        isAuthenticationError: false,
-      });
+    const { data, status } = await authenticateUser({ email, password }).catch(
+      () => {
+        this.setState({
```

Comment:
This code flow feels a bit weird. we can wrap the authenticateUser call in a normal try/catch so on success it'll go along with the rest of the checks you had below and then catch will be at the bottom